consensus/spos/sposFactory/sposFactory.go

```diff
@@ -29,6 +31,8 @@ func GetSubroundsFactory(
 		}
 
 		err = subRoundFactoryBls.SetAppStatusHandler(appStatusHandler)
+
+		subRoundFactoryBls.SetIndexer(indexer)
```

Comment:
move it before the SetAppStatusHandler or after the error handling

Reply:
Done.